### DISCUSSION TITLE
Questions about r and (r ...)

### DISCUSSION THREAD

**Discussion by FrauH0lle** [minad/tempel#156] [Q&A] [answered]
When I started using regions in my templates, I came up with some questions.

Tempo's documentation in `tempo-define-template` says about `r`:
`The symbol r: If tempo-insert is called with ON-REGION non-nil the current region is placed here. Otherwise it works like p.` 

Whereas Tempel's `r` states
 `If no region is active, quits the containing template when jumped to.` 
 
 I guess the different behavior is intentional thus, what would be the correct way to have a field in the template which is optionally a region and otherwise a normal placeholder?
 
```eld
(section-box
 >
 (p (if (string-suffix-p " " comment-start)
                          comment-start
                        (concat comment-start " ")) cs t)
 cs (make-string (- fill-column (current-indentation) (length cs)) ?-) n>
 cs (r "default title") n>
 cs (make-string (- fill-column (current-indentation) (length cs)) ?-) n>
 p n>
 q)
```
 
 When I use the template above, I get the default value when no region is active:
![grafik](https://github.com/user-attachments/assets/398d436d-2ae1-411f-9317-46703170f2f9)
However, in case there is a region
![grafik](https://github.com/user-attachments/assets/213e5f94-ff3e-4839-be93-427e993873e4)
I expected that the region replaces the default value. Is this intentional?

**Comment by minad** [comment, marked as answer]
> I guess the different behavior is intentional thus, what would be the correct way to have a field in the template which is optionally a region and otherwise a normal placeholder?

Most likely this feature doesn't exist.

> I expected that the region replaces the default value. Is this intentional?

Are you sure? I don't see this issue.
